controllers/azurecluster_controller_test.go

```diff
@@ -51,3 +56,63 @@ var _ = Describe("AzureClusterReconciler", func() {
 		})
 	})
 })
+
+func TestAzureClusterReconcilePaused(t *testing.T) {
```

Comment:
I originally tried to specify this test like the one above with envtest, but the race detector was flagging something in the logging calls whenever two objects were being created, even serially and with a 10s sleep in between. It was easier to reqork this test to use a fake client than try to fix the race which seemed unrelated to the rest of these changes. The test itself hasn't lost any fidelity during that rewrite AFACT.